### PR TITLE
Added lots of missing properties

### PR DESCRIPTION
**Pull request by Strato**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url https://github.com/Hacker0x01/react-datepicker/blob/master/docs/datepicker.md .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

File: react-datepicker/react-datepicker.d.ts [hunk 1]
```diff
@@ -6,14 +6,18 @@
 /// <reference types="react"/>
 
 declare module "react-datepicker" {
-    interface ReactDatePicker {
+    interface ReactDatePickerProps {
+        autoComplete?: string;
+        autoFocus?: boolean;
         className?: string;
+        customInput?: any
```
Comment (andrewBalekha) [inline react-datepicker/react-datepicker.d.ts]: as far as i know this should be a jsx type, not any

**Comment by Strato** [comment]
Done.

File: react-datepicker/react-datepicker.d.ts (hunk 1 above)
Comment (Strato) [inline react-datepicker/react-datepicker.d.ts]: Done.

**Comment by vvakame** [comment]
Can I merge this PR?

**Comment by andrewBalekha** [comment]
👍

**Pull request merged**